providers/websockets/binance/ws_data_handler.go

```diff
@@ -0,0 +1,166 @@
+package binance
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+
+	"github.com/skip-mev/slinky/oracle/config"
+	"github.com/skip-mev/slinky/oracle/types"
+	"github.com/skip-mev/slinky/providers/base/websocket/handlers"
+	"go.uber.org/zap"
+)
+
+var _ types.PriceWebSocketDataHandler = (*WebSocketHandler)(nil)
+
+// WebSocketHandler implements the WebSocketDataHandler interface. This is used to handle
+// messages received from the Binance websocket API.
+type WebSocketHandler struct {
+	logger *zap.Logger
+
+	// ws is the config for the Binance websocket.
+	ws config.WebSocketConfig
+	// cache maintains the latest set of tickers seen by the handler.
+	cache types.ProviderTickers
+	// messageIDs is the current message ID for the Binance websocket API per currency pair(s).
+	messageIDs map[int64][]string
+	// nextID is the next message ID to use for the Binance websocket API.
+	nextID int64
+}
+
+// NewWebSocketDataHandler returns a new Binance PriceWebSocketDataHandler.
+func NewWebSocketDataHandler(
+	logger *zap.Logger,
+	ws config.WebSocketConfig,
+) (types.PriceWebSocketDataHandler, error) {
+	if ws.Name != Name {
+		return nil, fmt.Errorf("expected websocket config name %s, got %s", Name, ws.Name)
+	}
+
+	if !ws.Enabled {
+		return nil, fmt.Errorf("websocket config for %s is not enabled", Name)
+	}
+
+	if err := ws.ValidateBasic(); err != nil {
+		return nil, fmt.Errorf("invalid websocket config for %s: %w", Name, err)
+	}
+
+	return &WebSocketHandler{
+		logger:     logger,
+		ws:         ws,
+		cache:      types.NewProviderTickers(),
+		messageIDs: make(map[int64][]string),
+		nextID:     rand.Int63() + 1,
```

Comment:
any reason not to just start it @ 1?

Reply:
since the connection can be multi-plexed we might duplicate the ID which might affect server side request handling.